Dockerfile

```diff
@@ -7,14 +7,11 @@ WORKDIR /community-server
 ## Copy the package.json for audit
 COPY package*.json ./
 
-## Verify if there are known vulnerabilities in the dependencies
-RUN npm audit --production --audit-level=high
-
```

Comment:
It is a bad idea to disable that step.
We need to update the dependencies so that they don't have vulnerabilities, not build containers with vulnerabilities.

Reply:
I'm not saying that this step should not executed, I'm just saying that this is not the correct place to run it.
We're also not executing the all unit+integration tests before a build, even though they must always pass.

This audit step should be called during CI, even _before_ Docker containers are being built.

It is not the responsibility for the Docker container builder to resolve vulnerability issues, this is something the package maintainer should take care off.

Because at the moment, no one is able to build Docker containers anymore, because there is an audit issue.

Reply:
> Dockerfile feels like a confusion of responsibilities

This is not mutually exclusive with having the audit gating  PRs to main.

Ultimately, the image will be build in CI. But having this step is a good practice safeguard for everyone building images in other ways.

Reply:
> Ultimately, the image will be build in CI.

Indeed, Docker images will be built in CI, together with all other CI steps (unit tests, linting, integration, ...).
So instead of running this audit during a Docker build, this should be run as a separate CI step.

> But having this step is a good practice safeguard for everyone building images in other ways.

Following that reasoning, we should also run all [CI steps](https://github.com/solid/community-server/blob/main/.github/workflows/ci.yml) during the Docker build, which is not the right approach IMO.

Reply:
Running tests during a docker build is not the same thing as preventing building containers that have vulnerabilities in their dependencies.

There are two scenarios:
1. The Docker container gets build from CI and the audit won't fail (otherwise it would upstream)
2. The Docker container gets build "in place X at instant Y" and will inform you the CSS is not fresh for creating an image, you can therefore investigate and fix instead of pushing an image that has vulnerabilities into your registry

More scenarios include any reason leading to the npm audit being disabled upstream (or ran in parallel isntead of upstream...).

All those circumstances justify in my opinion gating Docker builds with an audit. It will be easy for anyone with sufficient knowledge to bypass, but we wouldn't enable it.

Reply:
>  So instead of running this audit during a Docker build, this should be run as a separate CI step.

https://github.com/solid/community-server/issues/731